### PR TITLE
enable biblatex/biber

### PR DESCRIPTION
**Pull request by jakobjakobson13**

### CODE REVIEW

**Comment by jakobjakobson13** [comment]
Currently, I have two problems
- biber ends up at `/app/bin` (can be changed though). Not every editor has that directory in its PATH variable.
- If the directory `/app/bin` has been appended, I get the following error:
```
[📦 org.tug.texworks ~]$ biber
Can't locate strict.pm in @INC (you may need to install the strict module) (@INC contains: /usr/lib/sdk/texlive/lib/perl5/site_perl/5.34.0/x86_64-linux /usr/lib/sdk/texlive/lib/perl5/site_perl/5.34.0 /usr/lib/sdk/texlive/lib/perl5/5.34.0/x86_64-linux /usr/lib/sdk/texlive/lib/perl5/5.34.0) at /app/texlive/bin/biber line 4.
BEGIN failed--compilation aborted at /app/texlive/bin/biber line 4.
``` 
But even if I add `strict.pm`
```
[📦 org.tug.texworks ~]$ ls /app/texlive/lib/perl5/5.34.0/strict.pm 
/app/texlive/lib/perl5/5.34.0/strict.pm
```
to the PERL path
```
[📦 org.tug.texworks ~]$ set PERL5LIB=/app/texlive/lib/perl5/5.34.0/
```
I still get the same error
```
[📦 org.tug.texworks ~]$ biber
Can't locate strict.pm in @INC (you may need to install the strict module) (@INC contains: /usr/lib/sdk/texlive/lib/perl5/site_perl/5.34.0/x86_64-linux /usr/lib/sdk/texlive/lib/perl5/site_perl/5.34.0 /usr/lib/sdk/texlive/lib/perl5/5.34.0/x86_64-linux /usr/lib/sdk/texlive/lib/perl5/5.34.0) at /app/texlive/bin/biber line 4.
BEGIN failed--compilation aborted at /app/texlive/bin/biber line 4.
```
Can anyone with more perl experience give me any advice for that issue?

**Comment by fkastner** [comment]
How do you test this? `flatpak run` didn't work for me.
I then tested it with the texstudio flatpak. (for this I had to modify the `~/.local/share/flatpak/app/org.texstudio.TeXstudio/current/active/manifest` file to use `runtime=org.kde.Platform/x86_64/5.15-21.08` and in the section for the texlive extension set `version=21.08`)

Here `biber` ended up in `/app/texlive/bin/` and then it hit the same problem you described.
Part of the problem seems to be the wrong default path of perl. In your error message it says `@INC` contains some paths in `/usr/lib/sdk/texlive/lib/perl5/` but the actual paths are in `/app/texlive/lib/perl5/`.

Can you try `export PERL5LIB=/app/texlive/lib/perl5/5.34.0:/app/texlive/lib/perl5/site_perl/5.34.0/`?
For me this results in a new error message:
```
Can't load '/app/texlive/lib/perl5/site_perl/5.34.0/x86_64-linux/auto/Text/BibTeX/BibTeX.so' for module Text::BibTeX: libbtparse.so: cannot open shared object file: No such file or directory at /app/texlive/lib/perl5/5.34.0/x86_64-linux/DynaLoader.pm line 193.
 at /app/texlive/lib/perl5/site_perl/5.34.0/Biber.pm line 42.
```
It seems that it can't find `libbtparse.so` but I'm also not sure where that is/should be. I think it should get installed along with the perl Text-BibTeX package..

**Comment by hfiguiere** [comment]
Be aware that some packages just get the SDK instead of copy so you might want to kep `/usr/lib/sdk/texlive/` around in the path.

**Comment by jakobjakobson13** [comment]
> It seems that it can't find  but I'm also not sure where that is/should be. I think it should get installed along with the perl Text-BibTeX package..

`libbtparse.so` can be found at `/app/texlive/lib/libbtparse.so`. But that did not bring me any further. (I guess I'll go with my distribution latex packages again.)

File: org.freedesktop.Sdk.Extension.texlive.yml [hunk 1]
```diff
@@ -75,6 +75,48 @@ modules:
           url: https://pi.kwarc.info/historic/systems/texlive/2021/
           version-pattern: texlive-([\d\.]+).source.tar.xz
           url-template: https://pi.kwarc.info/historic/systems/texlive/2021/texlive-$version-tlpdb-full.tar.gz
+  - name: perl
+    no-autogen: true
+    config-opts:
+      --des
+      --Duseshrplib
```
Comment (hfiguiere) [inline org.freedesktop.Sdk.Extension.texlive.yml]:
```suggestion
      - -des
      - -Duseshrplib
```

This cause 
```
(flatpak-builder:2097273): Json-WARNING **: 18:36:35.977: Failed to deserialize "config-opts" property of type "GStrv" for an object of type "BuilderModule"
```

Comment (hfiguiere) [reply org.freedesktop.Sdk.Extension.texlive.yml]: but then with this change:
```
This GNU configure front end does not understand --des
```

Comment (hfiguiere) [reply org.freedesktop.Sdk.Extension.texlive.yml]: Nevermind updated the suggestion.

**Comment by hfiguiere** [comment]
With the changes suggested:

```
$ biber 
Usage:
      biber [options] file[.bcf]
      biber [options] --tool <datasource>

      Creates "file.bbl" using control file "file.bcf" (".bcf" extension is
      optional). Normally use with biblatex requires no options as they are
      all set in biblatex and passed via the ".bcf" file

      In "tool" mode (see B<--tool> option), takes a datasource (defaults to
      "bibtex" datasource) and outputs a copy of the datasource with any command-line
      or config file options applied.

      Please run "biber --help" for option details

```
so it doesn't error.

**Comment by hfiguiere** [comment]
Just to add how I did test this:
```
$ flatpak run --command=sh --runtime=org.freedesktop.Sdk org.freedesktop.Sdk.Extension.texlive
(sandbox) $ PATH=/usr/lib/sdk/texlive/bin:/usr/lib/sdk/texlive/bin/x86_64-linux/:$PATH
```

Note that testing with texstudio like that is unsupported. Better update its runtime.

File: org.freedesktop.Sdk.Extension.texlive.yml (hunk 1 above)
Comment (jakobjakobson13) [reply org.freedesktop.Sdk.Extension.texlive.yml]: @hfiguiere Thank you for your support. It really is a big help for me.

**Pull request merged**